src/features/houseHold/components/HouseholdGeneralForm.tsx

```diff
@@ -54,12 +58,27 @@ export function HouseholdForm({
     },
   });
 
-  function onSubmit(data: HouseholdSchema) {
+  async function onSubmit(data: HouseholdSchema) {
     if (household != null) {
-      updateHousehold(data, household.id);
-    } else {
-      createHousehold(data);
+      startTransition(async () => {
+        const result = await updateHousehold(data, household.id);
+
+        if (result.error) {
+          toast.error(result.message);
+          return;
+        }
+
+        toast.success(result.message);
+      });
+
+      return;
     }
+
+    startTransition(async () => {
+      await createHousehold(data);
+    });
```

Comment:
tego ifa można wstawić do jednego startTransition, wtedy error handling masz w jednym miejscu a nie rozdzielony (aktualnie jak wystąpi błąd tworzenia householda to nie będzie przez to toasta